src/PKSim.Core/Snapshots/Mappers/FormulationMapper.cs

```diff
@@ -1,22 +1,49 @@
-﻿using PKSim.Core.Model;
+﻿using OSPSuite.Core.Domain;
+using PKSim.Core.Model;
+using PKSim.Core.Repositories;
+using PKSim.Core.Services;
 using SnapshotFormulation = PKSim.Core.Snapshots.Formulation;
 using ModelFormulation = PKSim.Core.Model.Formulation;
 
 namespace PKSim.Core.Snapshots.Mappers
 {
    public class FormulationMapper : ParameterContainerSnapshotMapperBase<ModelFormulation, SnapshotFormulation>
    {
-      public FormulationMapper(ParameterMapper parameterMapper) : base(parameterMapper)
+      private readonly IFormulationRepository _formulationRepository;
+      private readonly ICloner _cloner;
+
+      public FormulationMapper(ParameterMapper parameterMapper, IFormulationRepository formulationRepository, ICloner cloner) : base(parameterMapper)
       {
+         _formulationRepository = formulationRepository;
+         _cloner = cloner;
       }
 
       public override SnapshotFormulation MapToSnapshot(ModelFormulation modelFormulation)
       {
          var snapshotFormulation = new SnapshotFormulation();
-         MapBaseProperties(modelFormulation, snapshotFormulation);
+         MapModelPropertiesIntoSnapshot(modelFormulation, snapshotFormulation);
          MapParameters(modelFormulation.AllVisibleParameters(), snapshotFormulation);
          snapshotFormulation.FormulationType = modelFormulation.FormulationType;
          return snapshotFormulation;
       }
+
+      public override ModelFormulation MapToModel(SnapshotFormulation formulationSnapshot)
+      {
+         var template = _formulationRepository.FormulationBy(formulationSnapshot.FormulationType);
+         var formulation = _cloner.Clone(template);
+
+         MapSnapshotPropertiesIntoModel(formulationSnapshot, formulation);
+         foreach (var snapshotParameter in formulationSnapshot.Parameters)
+         {
+            var modelParameter = formulation.Parameter(snapshotParameter.Name);
+
+            if (modelParameter == null)
```

Comment:
I think that if we are trying to load a snapshot with a parameter that we do not know, something is clearly outdated and we should throw an exception.
I don't think we should support project conversion for snapshot too. We need to keep them uptodate

Reply:
Exception or should we just write a warning into logfile and continue? I would vote for the warning.
At the end the user still can decide if the project may be used or should be thrown away.

And yes, would I also think we don't need project conversion for that.

Reply:
E.g. if starting in a batch mode.
If started in PK-Sim it might be not a log file but a summary (like in the old 4.2=>5.x project conversion)

Reply:
I think that loading something that does not match the template can only be wrong and lead to wrong result. How should a user know what to do?

Reply:
Besides those file should never be edited by hand in my opinion. Rather they should be generated from the tool

Reply:
I am going to create an entry for that to continue the discussion. We can always add this if we need

Reply:
ok